### PR TITLE
Function template for Redis/CosmosDB synchronization using StreamTrigger

### PR DESCRIPTION
**Pull request by phoebeowusu**
Function template for synchronizing **Redis to CosmosDB** using StreamTrigger. Save data in the cache using StreamTrigger and  transfer data from the cache to database in CosmosDB.

### CODE REVIEW

File: samples/dotnet/StreamTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            );
+
+            // Create new item using message information
+            CosmosItem item = new(
+                id: words[1],
+                values: vals
+            );
+
+            // Upload item into CosmosDB container
+            CosmosItem upsertedItem = await container.UpsertItemAsync<CosmosItem>(item);
```
Comment (royland123) [inline samples/dotnet/StreamTrigger.cs:52]: Variable assignment isn't necessary here. This line could just be "await container.UpsertItemAsync<CosmosItem>(item);"

Comment (phoebeowusu) [reply samples/dotnet/StreamTrigger.cs]: Fixed

File: samples/dotnet/StreamTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using System.Linq;
+using System.Collections.Generic;
+using System.Configuration;
+using StackExchange.Redis;
+using Microsoft.WindowsAzure.Storage.Queue.Protocol;
+using Microsoft.Azure.WebJobs;
```
Comment (prshri-msft) [inline samples/dotnet/StreamTrigger.cs:10]: nit: Sort usings. VS shortcut is Ctrl + R, Ctrl + G

File: samples/dotnet/StreamTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+            // Map each key value pair
+            Dictionary<string, string> dict = ParseResult(entry);
+            foreach (KeyValuePair<string, string> item in dict)
+            {
+                logger.LogInformation(item.Key + "=>" + item.Value.ToString());
+            }
+
+            // Insert into CosmosDB synchronously
+            Data sampleItem = new Data { id = entry.Id, values = dict };
```
Comment (prshri-msft) [inline samples/dotnet/StreamTrigger.cs:75]: This code is same as the the write behind till this point. Is it possible to extract this into a private helper method and then reuse it for both cases. DRY :)

File: samples/dotnet/StreamTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples
+{
+    public static class StreamTrigger
+    {
+        public const string localhostSetting = "REDIS_CONNECTION";
+        private static readonly IDatabase redisDB = ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable("REDIS_CONNECTION")).GetDatabase();
+
+        // Parser helper function for reading results - https://developer.redis.com/develop/dotnet/streams/stream-basics/#spin-up-most-recent-element-task
+        public static Dictionary<string, string> ParseResult(StreamEntry entry) => entry.Values.ToDictionary(x => x.Name.ToString(), x => x.Value.ToString());
```
Comment (dmonroym) [inline samples/dotnet/StreamTrigger.cs:20]: This function already exists in RedisUtilities.cs as StreamEntryToDictionary

File: samples/dotnet/StreamTrigger.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Write behind
+        [FunctionName(nameof(WriteBehindAsync))]
+        public static async Task WriteBehindAsync(
+                [RedisStreamTrigger(localhostSetting, "stream1")] StreamEntry entry,
```
Comment (dmonroym) [inline samples/dotnet/StreamTrigger.cs:25]: can you rename this to streamTest just to be consistent with the other samples?

File: samples/dotnet/StreamTrigger.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+{
+    public static class StreamTrigger
+    {
+        public const string localhostSetting = "REDIS_CONNECTION";
+        private static readonly IDatabase redisDB = ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable("REDIS_CONNECTION")).GetDatabase();
+
+        // Parser helper function for reading results - https://developer.redis.com/develop/dotnet/streams/stream-basics/#spin-up-most-recent-element-task
+        public static Dictionary<string, string> ParseResult(StreamEntry entry) => entry.Values.ToDictionary(x => x.Name.ToString(), x => x.Value.ToString());
+
+        // Write behind
```
Comment (dmonroym) [inline samples/dotnet/StreamTrigger.cs:22]: Add some logging info here and in write through

File: samples/dotnet/StreamTrigger.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        private static Data FormatData(StreamEntry entry, ILogger logger)
+        {
+            logger.LogInformation("ID: {val}", entry.Id.ToString());
+
+            // Map each key value pair
+            Dictionary<string, string> dict = ParseResult(entry);
+
+            foreach (KeyValuePair<string, string> item in dict)
+            {
+                logger.LogInformation(item.Key + " => " + item.Value);
```
Comment (dmonroym) [inline samples/dotnet/StreamTrigger.cs:61]: Is it necessary to log every key-value pair?

**Comment by dmonroym** [comment]
Please add some tests

**Pull request closed without merging**